presentation/src/main/java/com/bunbeauty/presentation/feature/additionlist/editadditionlist/EditAdditionViewModel.kt

```diff
@@ -0,0 +1,163 @@
+package com.bunbeauty.presentation.feature.additionlist.editadditionlist
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import com.bunbeauty.domain.exception.updateaddition.AdditionFullNameException
+import com.bunbeauty.domain.exception.updateaddition.AdditionNameException
+import com.bunbeauty.domain.exception.updateaddition.AdditionPriceException
+import com.bunbeauty.domain.model.addition.UpdateAddition
+import com.bunbeauty.domain.usecase.GetAdditionUseCase
+import com.bunbeauty.domain.usecase.UpdateAdditionUseCase
+import com.bunbeauty.presentation.extension.launchSafe
+import com.bunbeauty.presentation.viewmodel.base.BaseStateViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+private const val ADDITION_UUID = "additionUuid"
+
+@HiltViewModel
+class EditAdditionViewModel @Inject constructor(
+    private val savedStateHandle: SavedStateHandle,
+    private val getAdditionUseCase: GetAdditionUseCase,
+    private val updateAdditionUseCase: UpdateAdditionUseCase
+) : BaseStateViewModel<EditAddition.DataState, EditAddition.Action, EditAddition.Event>(
+    initState = EditAddition.DataState(
+        uuid = "",
+        name = "",
+        priority = 0,
+        price = null,
+        isLoading = true,
+        isVisible = false,
+        fullName = "",
+        hasEditFullNameError = false,
+        hasEditError = false,
+        hasEditNameError = false,
+        hasEditPriceError = false
+
+    )
+) {
+
+    override fun reduce(action: EditAddition.Action, dataState: EditAddition.DataState) {
+        when (action) {
+            is EditAddition.Action.OnBackClick -> addEvent { EditAddition.Event.Back }
+
+            EditAddition.Action.OnSaveEditAdditionClick -> updateEditAddition()
+
+            EditAddition.Action.InitAddition -> loadData()
+
+            is EditAddition.Action.OnVisibleClick -> setState {
+                copy(
+                    isVisible = action.isVisible
+                )
+            }
+
+            is EditAddition.Action.EditFullNameAddition -> setState {
+                copy(
+                    fullName = action.fullName
+                )
+            }
+
+            is EditAddition.Action.EditNameAddition -> setState {
+                copy(
+                    name = action.name
+                )
+            }
+
+            is EditAddition.Action.EditPriorityAddition -> setState {
+                copy(
+                    priority = action.priority.toIntOrNull() ?: 0
+                )
+            }
+
+            is EditAddition.Action.EditPriceAddition -> {
+                action.price.toIntOrNull()?.let { price ->
```

Comment:
Pls change it to
`price = action.price.toIntOrNull()`

Reply:
Fixed